test/test_tags_and_queries.jl

```diff
@@ -32,3 +33,52 @@ tag!(r[5], Int, 4, 5)
 
 @test querydelete!(r[2], :symbol1, 4, ❓) == Tag(:symbol1, 4, 5)
 @test querydelete!(r[2], :symbol1, 4, ❓) === nothing
+
+
+# tests for fifo and filo order queries
+# for RegRefs
+reg = Register(5)
+tag!(reg[3], EntanglementCounterpart, 2, 3)
```

Comment:
it is important for tests to be easy to read and understand. Here you have a very random looking order of entries, so it is very difficult for me to verify that your tests are actually correct. It would be much easier to have something like

```
tag!(reg[3], EntanglementCounterpart, 1, 2)
tag!(reg[3], EntanglementCounterpart, 3, 4)
tag!(reg[3], EntanglementCounterpart, 5, 6)
...
```

With repetitions or inversions or whatever else is necessary to make this tractable